zebra-network/src/peer_set/initialize.rs

```diff
@@ -315,74 +350,125 @@ where
                 .checked_sub(1)
                 .expect("the pool always contains an unresolved future") as f64
         );
-        // This is a little awkward because there's no select3.
-        match select(
-            select(demand_rx.next(), crawl_timer.next()),
-            handshakes.next(),
-        )
-        .await
-        {
-            Left((Left((Some(_demand), _)), _)) => {
+
+        let crawler_action = tokio::select! {
+            a = handshakes.next() => a.expect("handshakes never terminates, because it contains a future that never resolves"),
+            a = crawl_timer.next() => a.expect("crawl_timer never terminates"),
```

Comment:
For clarification maybe add `crawl_timer never terminates, because ...`

Reply:
```suggestion
            a = crawl_timer.next() => a.expect("timers never terminate"),
```